### PR TITLE
Update the documentation on events.

### PR DESCRIPTION
**Pull request by manthey**
Remove `type` as a field in events, as we usually use `event` (and `event` is always present).

Make some events more consistent.

This also updates documentation in typedef.js and object.js.

### CODE REVIEW

**Comment by manthey** [comment]
@jbeezley Is there a better way to show the data that is passed to an event?  Currently, the geo.event lists some properties at the top, then each event lists properties specific to that event; it may not be obvious to the new user that the `event` and `geo` properties always exist.

Also, the geo.event.feature events should reference the main events, but I'm not sure how to get jsdoc to create the appropriate link.

**Comment by jbeezley** [comment]
I don't have any specific advice unfortunately.  I suspect we will need to customize the template or create a jsdoc plugin.  I can spend some time on it in a different PR.

**Comment by manthey** [comment]
This is a question for future documentation work:  The map class and some features, such as contour, take an object with many parameters.  I think these really should be specified using `@typedef` (currently, in map.js it is just listed as a long set of options in the class specification)?   If we do so, should the typedefs be moved to `typedef.js` or left in the appropriate feature?  In the map, this means that the class specification would just be `@param {geo.mapSpec} arg The map specification object`.

File: src/event.js [hunk 1]
```diff
@@ -3,15 +3,16 @@
  * Common object containing all event types that are provided by the GeoJS
  * API.  Each property contained here is a valid target for event handling
  * via {@link geo.object#geoOn}.  The event object provided to handlers is
- * different for each event type.  Each handler will generally be called
- * with a the <code>this</code> context being the class that caused the event.<br>
+ * different for each event type.  Each handler is generally be called
```
Comment (jbeezley) [inline src/event.js]: `s/be //`

File: src/event.js [hunk 2]
```diff
@@ -3,15 +3,16 @@
...
+ * with the <code>this</code> context being the class that caused the event.<br>
```
Comment (jbeezley) [inline src/event.js]: Might as well just use markdown syntax instead of `<code>`.

File: src/event.js [hunk 3]
```diff
@@ -167,25 +125,29 @@ geo_event.draw = 'geo_draw';
...
 //////////////////////////////////////////////////////////////////////////////
 geo_event.mousemove = 'geo_mousemove';
 
 //////////////////////////////////////////////////////////////////////////////
 /**
- * Triggered on every 'mousedown' over the map's DOM element.  The event
- * object extends {@link geo.mouseState}.
- * @mixes geo.mouseState
+ * Triggered on `mouseup` events that happen soon enough and close enough to a
+ * mousedown event.  The event object extends {@link geo.mouseState}.
```
Comment (jbeezley) [inline src/event.js]: <pre>`mousedown`</pre>

File: src/event.js [hunk 4]
```diff
@@ -404,31 +422,40 @@ geo_event.feature = {
...
 geo_event.pixelmap = {
-  /* The image associated with the pixel map url has been prepared and rendered
-   * once. */
+  /**
+   * Report that an image associated with a pixel map has been prepared and
+   * rendered once.
+   *
+   * @event geo.event.pixelmap.prepared
+   * @type {object}
+   * @property {geo.pixelmapFeature} pixelmap The pixelamp object that was
```
Comment (jbeezley) [inline src/event.js]: pixelmap

File: src/typedef.js [hunk 5]
```diff
@@ -81,44 +89,124 @@
...
- * brush action
+ *      brush action.
+ */
+
+/**
+ * The conditions that are necessary to make an action occur.
+ *
+ * @typedef geo.actionRecord
+ * @type {object}
+ * @property {string} action The name of the action, from geo.action.
```
Comment (jbeezley) [inline src/typedef.js]: `{@link geo.action}`

File: src/typedef.js [hunk 6]
```diff
@@ -81,44 +89,124 @@
...
+ * @property {string} [name] A human-readable name that can be used to locate
+ *      or filter actions.
+ * @property {string|object} input The name of an input that is used for the
+ *      action, or an object with input names as keys and boolean values of
+ *      inputs that are required to occur or required to not occur to trigger
+ *      the action.  Input names include `left`, `right`, `middle` (for mouse
+ *      buttons), `wheel` (the mouse wheel), `pan` (touch pan), `rotate` (touch
+ *      rotate).
+ * @property {string|object} [modifiers] The name of a modifier key or an
+ *      object withe modifiers as the keys and boolean values.  The listed
```
Comment (jbeezley) [inline src/typedef.js]: typo

File: src/event.js [hunk 7]
```diff
@@ -194,7 +156,8 @@ geo_event.mouseclick = 'geo_mouseclick';
 /**
  * Triggered on every 'mousemove' during a brushing selection.
```
Comment (jbeezley) [inline src/event.js]:
```
`mousemove`
```

**Pull request merged**